### PR TITLE
Implemented blog post edit

### PR DESCRIPTION
**Pull request by Naztarr**
#### This pull request features schema creation for the blog entity and service implementation of the blog post edit by superadmin:

```CREATE TABLE blogs (
    id BIGINT PRIMARY KEY,
    blog_id VARCHAR(36) UNIQUE,
    title VARCHAR(100) NOT NULL,
    content TEXT NOT NULL,
    image_urls TEXT[],
    tags TEXT[],
    author VARCHAR(255),
    created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
    updated_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP
);

-- Create trigger function for update column
CREATE OR REPLACE FUNCTION update_updated_at_column()
RETURNS TRIGGER AS $$
BEGIN
    NEW.updated_at = CURRENT_TIMESTAMP;
    RETURN NEW;
END;
$$ LANGUAGE plpgsql;

-- Create trigger to call the function before any update
CREATE TRIGGER set_updated_at
BEFORE UPDATE ON blogs
```
#### dto and mapper classes were created to map inputs to the database after proper verification and formatting

 ```
├── blog
│   │   │       │   ├── controller
│   │   │       │   │   └── BlogController.java
│   │   │       │   ├── dto
│   │   │       │   │   ├── EditRequest.java
│   │   │       │   │   └── EditResponse.java
│   │   │       │   ├── entity
│   │   │       │   │   └── Blog.java
│   │   │       │   ├── mapper
│   │   │       │   │   └── BlogMapper.java
│   │   │       │   ├── repository
│   │   │       │   │   └── BlogRepository.java
│   │   │       │   └── service
│   │   │       │       ├── EditBlogImplementation.java
│   │   │       │       └── EditBlogService.java
```

spring-boot-starter-validation was added to handle validation at controller and dto levels

### CODE REVIEW

**Comment by NdubuisiJr** [comment]
@Naztarr Add proper description of what you have done. All your build failed.

File: src/main/java/hng_java_boilerplate/blog/controller/BlogController.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api/v1/blog")
+public class BlogController {
+    private final EditBlogService editBlogService;
+
+
+    @PutMapping("/edit/{blog_id}")
+    public ResponseEntity<ApiResponse<EditResponse>> edit_blog(@PathVariable String blog_id,
+                                                               @RequestBody EditRequest request){
```
Comment (billmal071) [inline src/main/java/hng_java_boilerplate/blog/controller/BlogController.java:20]: Add validations

File: src/main/java/hng_java_boilerplate/blog/service/EditBlogImplementation.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+import java.util.*;
+import java.util.regex.Pattern;
+
+@Service
+@RequiredArgsConstructor
+public class EditBlogImplementation implements EditBlogService {
+    private final BlogRepository blogRepository;
+    private final Pattern TAG_PATTERN = Pattern.compile("^[a-zA-Z0-9-_ ]+$");
+    private final int MAXIMUM_TAG_LENGTH = 30;
```
Comment (billmal071) [inline src/main/java/hng_java_boilerplate/blog/service/EditBlogImplementation.java:22]: Make these static

**Review by billmal071 — CHANGES_REQUESTED** [review]

File: src/main/java/hng_java_boilerplate/blog/service/EditBlogImplementation.java (hunk 2 above)
Comment (Naztarr) [reply src/main/java/hng_java_boilerplate/blog/service/EditBlogImplementation.java]: done

**Pull request closed without merging**